### ISSUE TITLE
[DOCS]: Adding clarification about when single mode layers would naturally also work with disjoint mode.

### ISSUE THREAD

**Issue by LeviBorodenko** [danielegrattarola/spektral#57]
I believe that the way single and disjoint mode are handled in the documentation is potentially confusing.

For example, `SortPool` works in single mode but does not work in disjoint mode.

Meanwhile if we look at  `GraphConv`, the documentation says that it works for single mode but does not mention disjoint mode. This gives the impression that maybe this layer is also incompatible with disjoint mode. However, unless I am misunderstanding something, `GraphConv` works fine with disjoint graphs.

Therefore, I would suggest to exclude disjoint mode only if the layer will not work properly on disjoint graphs.

Best,
Levi

**Comment by danielegrattarola** [comment]
Yeah, I agree that the whole single/disjoint thing can be a bit confusing. 
I can add explicit disjoint support in the docs of all convolutional layers that support it, although I think that the whole modes system should be re-thought and improved.

**Comment by LeviBorodenko** [comment]
That be great!

Another issue that I am running into is, that disjoint mode causes problems with graph level classification tasks. For instance, `keras.Model.fit` requires a label for each member of the batch, thus making it impossible(?) to use anything but node-level labels.

I think it may be wise to give disjoint graphs a trivial batch dimension instead of using the node dimension as the batch dimension. Thus the signals would have shape [1, nodes, features]. This would fix issues like having to disable shuffling, having to set batch-size to be the number of nodes and not being able to use graph-level labels.

Best,
Levi

**Comment by danielegrattarola** [comment]
@LeviBorodenko sorry, I completely forgot about this issue. 
The problem with doing this reshape that you mention is that it would introduce additional complexity and I'm not sure that it would improve usability. 
Also, there is no support for sparse matrices of rank 3 in Scipy, and terrible support in TensorFlow.

I agree that it would be better if we could do graph-level training using the Keras `fit()` loop, but I don't think that adding a dummy batch size is the solution here. 
A custom training loop [like this](https://github.com/danielegrattarola/spektral/blob/master/examples/graph_prediction/qm9_disjoint.py) is less hacky and also faster in training.

Cheers, 
Daniele
